src/sources/internal_metrics.rs

```diff
@@ -93,6 +109,12 @@ async fn run(
         let metrics = capture_metrics(controller);
 
         out.send_all(&mut stream::iter(metrics).map(|mut metric| {
+            // A metric starts out with a default "vector" namespace, but will be overridden
+            // if an explicit namespace is provided to this source.
+            if namespace.is_some() {
+                metric = metric.with_namespace(namespace.as_ref());
+            }
```

Comment:
This is kinda making me wonder if metrics should be published without a namespace and only set the namespace here.

Reply:
🤔  Yeah, that's an option. The scope of that decision is bigger than this PR, but something to consider.